src/platform/forms-system/test/js/utilities/file.unit.spec.js

```diff
@@ -0,0 +1,61 @@
+import { expect } from 'chai';
+
+import { checkForEncryptedPdf } from 'platform/forms-system/src/js/utilities/file';
+
+describe('checkForEncryptedPdf', () => {
+  let oldFileReader;
+
+  const setup = (isEncrypted, ext = 'pdf') => {
+    oldFileReader = global.FileReader;
+    global.FileReader = function() {
+      this.readAsArrayBuffer = () => {};
+
+      const result = isEncrypted
+        ? [0, 0, ...'/Encrypt'.split('').map(s => s.charCodeAt(0)), 0, 0]
+        : new Array(20).fill(0);
+
+      const onloadendEvent = {
+        target: {
+          readyState: 'done',
+          result,
+        },
```

Comment:
Why is this mocking happening? I'm a bit nervous about doing this much mocking for a test.

Reply:
I'm mocking the binary data of an encrypted PDF file. It includes an `/Encrypt` string within the file, and it's the only way I know how to detect if a PDF is encrypted.

I could have used `readAsBinaryString` instead of `readAsArrayBuffer` to simplify the `checkForEncryptedPdf` code, but [we support IE11](https://developer.mozilla.org/en-US/docs/Web/API/FileReader#Browser_compatibility) which just happens to not support `readAsBinaryString`.